pkg/generate/app/cmd/newapp.go

```diff
@@ -314,10 +348,13 @@ func (c *AppConfig) Run(out io.Writer) (*AppResult, error) {
 	glog.V(4).Infof("Code %v", repositories)
 	glog.V(4).Infof("Images %v", components)
 
-	if err := c.detectSource(repositories); err != nil {
+	srcComponents, err := c.detectSource(repositories)
+	if err != nil {
```

Comment:
I think this is too late - we want to use source to suggest searchable terms that we then resolve, rather than doing resolution from within the source detection.  It requires more dramatic changes to this function though - moving detection up earlier.  If users can tie an input to a source, then it can be after (and we just warn that the source doesn't match the input ref), but if we truly want a source repo to suggest more complex setup we would need to do it as a pre-processing step before we calculate the resolved images.

Also, if the repository is already linked to a component, we shouldn't be adding another one here (regardless of the above comment).